app/src/main/java/ch/epfl/sdp/appart/login/FirebaseLoginService.java

```diff
@@ -44,6 +44,7 @@ public static LoginService buildLoginService() {
 
     @Override
     public CompletableFuture<User> loginWithEmail(String email, String password) {
+        if (getCurrentUser() == null) throw new IllegalStateException("current user must not already be set when authentication");

```

Comment:
shouldn't the user be null if we're trying to log in?

Reply:
I agree with this.

Reply:
yep this is inverted